crates/driver/src/lib.rs

```diff
@@ -6,22 +6,71 @@ pub use db::{DriverDataBase, DriverDb};
 
 use clap::{Parser, Subcommand};
 use hir::hir_def::TopLevelMod;
+use resolver::{
+    ingot::{
+        config::ConfigResolver,
+        source_files::{self, SourceFilesResolver},
+    },
+    Resolver,
+};
 
 pub fn run(opts: &Options) {
     match &opts.command {
         Command::Build => eprintln!("`fe build` doesn't work at the moment"),
-        Command::Check { path } => {
+        Command::Check { path, core } => {
             if !path.exists() {
-                eprintln!("file '{}' does not exist", path);
+                eprintln!("The path '{}' does not exist", path);
                 std::process::exit(2);
             }
-            let source = std::fs::read_to_string(path).unwrap();
 
             let mut db = DriverDataBase::default();
-            let (ingot, file) = db.standalone(path, &source);
-            let top_mod = db.top_mod(ingot, file);
-            let diags = db.run_on_top_mod(top_mod);
-            diags.emit(&db);
+            let mut config_resolver = ConfigResolver::default();
+            let mut source_files_resolver = SourceFilesResolver::default();
+
+            if let Some(core_path) = core {
+                let config = config_resolver
+                    .resolve(path)
+                    .expect("local config resolution failed");
+                dbg!(config_resolver.take_diagnostics());
+
+                let source_files = source_files_resolver
+                    .resolve(path)
+                    .expect("local source files resolution failed");
+                dbg!(source_files_resolver.take_diagnostics());
+
+                let (input_ingot, source_files) = db.local_ingot(
+                    path,
+                    config.version.expect("local config version is empty"),
+                    &source_files.files,
+                );
+            } else {
+                todo!("static loading")
+            };
+
+            if path.is_dir() {
+                let config = config_resolver
+                    .resolve(path)
+                    .expect("local config resolution failed");
+                dbg!(config_resolver.take_diagnostics());
+
+                let source_files = source_files_resolver
+                    .resolve(path)
+                    .expect("local source files resolution failed");
+                dbg!(source_files_resolver.take_diagnostics());
+
+                let (input_ingot, source_files) = db.local_ingot(
+                    path,
+                    config.version.expect("local config version is empty"),
+                    &source_files.files,
+                );
+            } else {
+                let source = std::fs::read_to_string(path).unwrap();
+
+                let (ingot, file) = db.standalone(path, &source);
+                let top_mod = db.top_mod(ingot, file);
+                let diags = db.run_on_top_mod(top_mod);
+                diags.emit(&db);
+            }
```

Comment:
collect diagnostics for all modules in core, local, and standalone ingots and print